src/Orleans.Runtime/GrainDirectory/LocalGrainDirectory.cs

```diff
@@ -792,10 +773,18 @@ public bool LocalLookup(GrainId grain, out AddressesAndTag result)
         {
             localLookups.Increment();
 
-            SiloAddress silo = CalculateTargetSilo(grain, false);
-            // No need to check that silo != null since we're passing excludeThisSiloIfStopping = false
+            SiloAddress silo = CalculateGrainDirectoryPartition(grain);
+
 
-            if (log.IsEnabled(LogLevel.Debug)) log.Debug("Silo {0} tries to lookup for {1}-->{2} ({3}-->{4})", MyAddress, grain, silo, grain.GetUniformHashCode(), silo.GetConsistentHashCode());
+            if (log.IsEnabled(LogLevel.Debug)) log.Debug("Silo {0} tries to lookup for {1}-->{2} ({3}-->{4})", MyAddress, grain, silo, grain.GetUniformHashCode(), silo?.GetConsistentHashCode());
+
+            //this will only happen if I'm the only silo in the cluster and I'm shutting down
+            if (silo == null)
+            {
+                if (log.IsEnabled(LogLevel.Trace)) log.Trace("LocalLookup mine {0}=null", grain);
+                result = new AddressesAndTag();
```

Comment:
Should this be null instead?

Reply:
its caller expects it not be null. And also see line 808 and line 811. it is doing the same thing. I was copying logic from there.